### ISSUE TITLE
Tabbing right when in the last column of a table adds a dangling whitespace

### ISSUE THREAD

**Issue by luckman212** [tgrosinger/advanced-tables-obsidian#61]
Not sure if this can be avoided... but it triggers a [MD009](https://github.com/DavidAnson/markdownlint/blob/v0.22.0/doc/Rules.md#md009) warning in `markdownlint` and I usually end up hunting and whack-a-mole'ing those extra spaces.

Here's a video... this is Obs 0.10.13 + plugin v.0.10.0
![CleanShot 2021-02-07 at 13 55 00](https://user-images.githubusercontent.com/1992842/107156496-aadd6880-694c-11eb-8107-172b11f84628.gif)

I think a nice way to handle this would be to have an option that <kbd>TAB</kbd> from the rightmost column wraps around to the first column on the next row instead of creating a new column. Inserting new columns could be triggered only when explicitly commanded to do so via the toolbar button or hotkey.

**Comment by tgrosinger** [comment]
The reason it does this is to set you up for writing a new column. If you were to put new text there and press enter it would add the trailing pipe character and you would have 4 columns in the table. This is a very smooth method of building the table up from scratch since you can just press tab repeatedly.

This gif demos that workflow:

![tab-creating-table](https://raw.githubusercontent.com/tgrosinger/advanced-tables-obsidian/main/resources/screenshots/basic-functionality.gif)

**Comment by luckman212** [comment]
@tgrosinger Yes of course, I get that. It 100% makes sense and speeds entry when tabbing through cells of an existing table, adding rows, etc.

But as far as adding new columns when tabbing from the _rightmost_ cell, not sure if that's typically the desired outcome. Maybe I'm the edge case, but usually I know what my columns (aka fields) are when setting up the table. It's rare that I want to add additional fields when just tabbing around. 

Most of the time my intention is for <kbd>TAB</kbd> to advance me to the first column of the next line (creating a new row if needed) vs. adding an entire new column. For the rare cases when I actually need a new column, I'd just use the add column toolbar button.

Not sure if this could be made into an option, or if there could somehow be logic added such that 

```
IF we are in the top (header) row
AND tab is pressed
THEN add a new column
ELSE skip to next line
```

**Comment by tgrosinger** [comment]
Hmm, I see what you're saying. That would be possible, but currently the plugin doesn't have enough information to know if you are past in the last column. I'll have to make a change to the underlying library which actually does the markdown table work (this plugin is just an interface between that library and Obsidian).

Next time I'm working on that library I'll work on adding this 🙂

**Comment by chrisl64squares** [comment]
Just noting that this would be a welcome addition! I'd rather tabbing from the last column did nothing than add a new column. Ideally it would know if there is an existing next row and go to its first column or, if not, add a row. Or something 😄
